### ISSUE TITLE
Do not emit CA2213 when the field is disposed in DisposeCore method

### ISSUE THREAD

**Issue by SergeyTeplyakov** [dotnet/roslyn-analyzers#3252]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.8 (Latest)

#### Diagnostic ID

CA2213

#### Repro steps

```csharp
public class Disposable : IDisposable
    {
        // Some extra checks to allow double dispose
        public void Dispose() => DisposeCore();
        protected virtual void DisposeCore() { }
    }

    public class FooBar : Disposable
    {
        private readonly IDisposable _f = Stream.Null;

        protected override void DisposeCore() => _f.Dispose(); // CA2213
    }
```

#### Expected behavior
In the current project, we're using a custom version of the Dispose pattern. We can even debate about the value of the standard one, but this is IMO slightly out of scope for this issue.

The question is, is it possible to "tell" the analyzer somehow that our "DisposeCore" method is part of the resource disposal procedure and avoid this warning in all the cases?

Right now we still could suppress the warning manually in the source or globally, but I would actually prefer a solution when the analyzer can recognize a custom dispose patterns instead.

**Comment by Evangelink** [comment]
@mavasani is dataflow execution supporting inter-procedural calls? If it does we shall be able to get rid of this in a quite generic way.
